src/stream/buf_reader.rs

```diff
@@ -338,17 +348,23 @@ fn tokio_read_buf(
         bs.reserve(8 * 1024);
     }
 
-    unsafe {
-        let uninit = bs.chunk_mut();
-        let mut buf = tokio_dep::io::ReadBuf::uninit(std::slice::from_raw_parts_mut(
+    let uninit = bs.chunk_mut();
+
+    // SAFETY: no one else has pointers into the buffer right now
```

Comment:
All of these comments about the only mutable reference feels redundant. Mutability doesn't factor in, the unsafe is only needed to convert bytes uniitialized bytes type into the one that ReadBuf expects.

```suggestion
    // SAFETY: Only converting one uninitialized buffer to another type 
```

perhaps

Reply:
I think this is resolved by rebasing onto #317.